src/Utilities/Compiler/Options/EditorConfigOptionNames.cs

```diff
@@ -204,5 +204,16 @@ internal static partial class EditorConfigOptionNames
         /// CA1826 (Do not use Enumerable methods on indexable collections. Instead use the collection directly).
         /// </summary>
         public const string ExcludeOrDefaultMethods = "exclude_ordefault_methods";
+
+        /// <summary>
+        /// String option to configure names of method symbols (separated by '|') that marks all of the parameters with IEnumerable type
+        /// would be enumerated.
+        /// </summary>
+        public const string EnumerationMethods = "enumeration_methods";
```

Comment:
Are we not going to have a boolean option that turns our current conservative heuristic `assume that by default method invocations that take IEnumerable parameters do not enumerate the argument` into the aggressive one `assume that method invocations that take IEnumerable parameters enumerate the argument`? Basically, flipping this bool option would allow repos to do a code cleanup pass where they go aggressive to catch more potential violations (ignoring false positives), but then flip the option back off so the CI is still running in conservative mode. This might give us best of both worlds.

I'll let you and @mikadumont decide if this is a valuable option, and if valuable, whether you want to do it in this PR or a follow-up change.

Reply:
If we want to make it configurable, then we will
1. A boolean value configures the default behavior. (True to be more aggressive, false to conservative)
2. Another option called 'NoEnumerationMethods' to configure 'NoEnumerationMethods'
Technically, we could reuse 'EnumerationMethods' option here and filp it everytime the default value is changed.
But I personally feel it becomes hard to understand.

Reply:
If we use such boolean value, probably no need to add other string options for method names, as users can flip it back amd forth as needed, just like @mavasani's usage example

Reply:
I see, `EnumerationMethods` option would not be used (because the default behavior has already done that)
And in that mode, we don't allow users to specify  `NoEnumerationMethods`, is that correct?

This would be a simplified solution.

Reply:
But in any case, I feel it can be done in the next PR, don't want to make this PR becomes overwhelming